src/puppetlabs/puppetdb/query.clj

```diff
@@ -422,21 +434,23 @@
           (:where %)]}
   (match [path]
          ["tag"]
-         {:where (sql-regexp-array-match "catalog_resources" "catalog_resources" "tags")
+         {:where (.s (sql-regexp-array-match "catalog_resources"
```

Comment:
This causes a reflection warning and I was confused as to what a .s was. There's a similar function that converts a SqlRaw into a keyword, I think a type hinted function there would make this clearer and get rid of the reflection.